### PR TITLE
chore(zone-sdk): simplify reorg event

### PR DESCRIPTION
**Pull request by pradovic**
## 1. What does this PR implement?

This simplifies the `ChannelUpdate` event, reducing to only { orphaned , adopted }, deduped by pending tx. 

It also simplifies the re-submission policies and adds another e2e test for concurrent sequencers. Also adds more e2e tests to cover new scenarios.

Also, now multiple sequencers with same signing keys are supported, so the orphaned and adopted are filtered per pending tx set and not per signing key.

## 2. Does the code have enough context to be clearly understood?

Yes

## 3. Who are the specification authors and who is accountable for this PR?

@pradovic @davidrusu 

## 4. Is the specification accurate and complete?
Yes

## 5. Does the implementation introduce changes in the specification?

No, the spec for FFW is still in progress.

## Checklist

> [!WARNING]  
> Do not merge the PR if any of the following is missing:

* [ ] 1. The PR title follows the Conventional Commits [specification](https://www.notion.so/How-to-open-a-Pull-Request-215261aa09df80deb538ed59f5e4e0b5).
* [ ] 2. Description added.
* [ ] 3. Context and links to Specification document(s) added.
* [ ] 4. Main contact(s) (developers and specification authors) added
* [ ] 5. Implementation and Specification are 100% in sync including changes. This is critical.
* [ ] 6. Link PR to a specific milestone.
* [ ] 7. New or changed logs were reviewed against the [logging guidelines](https://github.com/logos-blockchain/logos-blockchain/blob/master/CONTRIBUTING.md#logging-guidelines).

### CODE REVIEW

File: tests/src/tests/zone_sdk/e2e.rs [hunk 1]
```diff
@@ -1045,6 +1037,252 @@ async fn test_sorted_conflict_resolution() {
...
+    publish_concurrently(vec![
+        (handle_a, payloads_a.clone()),
+        (handle_b, payloads_b.clone()),
+        (handle_c, payloads_c.clone()),
+    ])
+    .await;
+
+    // Wait for things to settle. With k=10 and 1s slots, finality is ~10s
+    // per block. We wait long enough for retry cycles + finality.
+    sleep(Duration::from_secs(180)).await;
```
Comment (hansieodendaal) [inline tests/src/tests/zone_sdk/e2e.rs]: I think "Wait for things to settle" can be more deterministic, as each sequencer will report events detailing state.

File: tests/src/tests/zone_sdk/e2e.rs [hunk 2]
```diff
@@ -769,14 +755,14 @@ async fn test_concurrent_multi_sequencer() {
...
 
-    wait_for_indexer_unordered(&indexer, &expected_all, Duration::from_mins(20)).await;
+    wait_for_indexer_unordered(&indexer, &expected_all, Duration::from_mins(30)).await;
 
     // Wait for enough blocks so any late re-published duplicates would have
-    // landed. With k=3 and 1s slots, finality is ~3 blocks. We wait 30s
+    // landed. With k=10 and 1s slots, finality is ~10 blocks. We wait 60s
     // to be safe — enough for resubmit cycles and in-flight txs to settle.
-    sleep(Duration::from_secs(30)).await;
+    sleep(Duration::from_secs(60)).await;
```
Comment (hansieodendaal) [inline tests/src/tests/zone_sdk/e2e.rs]: ditto - could this not be deterministic?

File: testnet/tui-zone/src/state.rs [hunk 3]
```diff
@@ -105,39 +90,25 @@ impl ZoneState for InMemoryZoneState {
...
+/// track current canonical membership. Once a message is added it stays;
+/// there is no revert. Reorgs and bouncing don't change what was published,
+/// only where it lives on chain.
 ///
-/// Authorship is read from `state.is_ours`, which is durable across reorgs —
-/// so the order is the natural one: mutate state first, then decide based on
-/// the new chain.
+/// 1. Apply adopted to state (others' new inscriptions).
+/// 2. Iterate orphaned and return entries not in `pending` — those are
+///    republish candidates the user must decide on.
```
Comment (hansieodendaal) [inline testnet/tui-zone/src/state.rs]: I think for this TUI demo we can assert that every orhaned entry carries our signature

File: tests/src/tests/zone_sdk/e2e.rs [hunk 4]
```diff
@@ -291,48 +291,26 @@ fn spawn_drive(
...
-                    && !state.contains(&inv.payload)
-                    && !pending_payloads.contains(&inv.payload)
-                {
-                    debug!(
-                        "Re-publishing invalidated: {:?}",
-                        String::from_utf8_lossy(&inv.payload)
-                    );
+            for inv in &orphaned {
+                if !pending_payloads.contains(&inv.payload) {
+                    debug!("Re-publishing: {:?}", String::from_utf8_lossy(&inv.payload));
```
Comment (hansieodendaal) [inline tests/src/tests/zone_sdk/e2e.rs]: We should assert all orphaned carries our signature

**Review by hansieodendaal — COMMENTED** [review]
Nice simplification and looking good overall. Possible areas for improvement are:

1. Assert ownership of orphaned explicitly
   Since the new contract is that orphaned only contains entries that belong to this sequencer, I think both the E2E helpers and the TUI demo should assert that invariant explicitly by checking `InscriptionInfo.signer == our_public_key`. That makes the new API contract executable and protects us against future regressions.

1. Replace magic waits with event-driven settling
   I think all the “wait for things to settle” type sleeps should be made more deterministic. Each sequencer already emits enough state-transition events to drive convergence checks, so we should prefer waiting on explicit conditions over fixed sleep(...) calls.

1. Add identical-payload concurrent conflict test
   We should add a concurrent competing-sequencer test where all payloads are identical. Right now most concurrent tests rely on unique payloads, but identical payloads are a useful edge case for dedup/conflict/republish behavior and would better exercise the simplified orphaned semantics.

File: tests/src/tests/zone_sdk/e2e.rs [hunk 5]
```diff
@@ -1045,6 +1141,383 @@ async fn test_sorted_conflict_resolution() {
...
+    handle_a.wait_ready().await;
+    handle_b.wait_ready().await;
+    handle_c.wait_ready().await;
+
+    publish_concurrently(vec![
+        (handle_a, vec![payload.clone()]),
+        (handle_b, vec![payload.clone()]),
+        (handle_c, vec![payload.clone()]),
+    ])
+    .await;
```
Comment (hansieodendaal) [inline tests/src/tests/zone_sdk/e2e.rs:1415]: For this test to be effective, each sequencer client should publish multiple of the same, say 10 each, otherwise conflict resolution will not be exercised adequately.

All else looking good.

Comment (pradovic) [reply tests/src/tests/zone_sdk/e2e.rs:1415]: Ah ok I see. I thought this is a know limitation of this approach, as the user cannot really distinguish the tx if they are all the same and make a decision weather to resubmit, but let me try, it is a good use case to document a justify more complex lineage tracking if needed. This is why we have tx_uuud in tui.

Comment (pradovic) [reply tests/src/tests/zone_sdk/e2e.rs:1415]: Edit: @hansieodendaal proposed to use inscription info to make a policy, I will try

**Comment by pradovic** [comment]
@hansieodendaal @davidrusu 

Regarding adopted filter — I think this is a design call.

Today in `build_channel_event`:

- `adopted` = block-delta inscriptions on the canonical branch, filtered to exclude signer == my_signer because our own publishes are already delivered via Event::Published.
- `pending / orphaned` = derived from our local pending tx state, not from signer filtering, so these semantics would stay the same.

The current filter gives us a simple “is this mine?” signal under the one-key-per-sequencer assumption.

To support shared signing keys across multiple instances, we would need to drop the adopted signer filter. Then applications would need to maintain a local `HashSet<MsgId>` from `Event::Published` and dedup adopted against it to avoid double-counting their own messages.

I lean toward keeping the filter and explicitly not supporting shared keys for now. Multiple sequencers feels like a decentralization pattern — different signers per instance — not horizontal scaling under one identity. I can add a doc note making that assumption explicit.

WDYT?

File: tests/src/tests/zone_sdk/e2e.rs (hunk 5 above)
Comment (pradovic) [reply tests/src/tests/zone_sdk/e2e.rs:1415]: Done

**Comment by hansieodendaal** [comment]
> @hansieodendaal @davidrusu
> 
> Regarding adopted filter — I think this is a design call.
> 
> Today in `build_channel_event`:
> 
> WDYT?

Does a client not know the tx_id of their own submitted inscription? Then a shared signing key should not be a problem, and the adopted filter can be kept.

**Comment by pradovic** [comment]
> > @hansieodendaal @davidrusu
> 
> > 
> 
> > Regarding adopted filter — I think this is a design call.
> 
> > 
> 
> > Today in `build_channel_event`:
> 
> > 
> 
> > WDYT?
> 
> 
> 
> Does a client not know the tx_id of their own submitted inscription? Then a shared signing key should not be a problem, and the adopted filter can be kept.

It can, the info is there but it complicates the tui implementation to cover this case. Also, I see several problems there:
1. Adopted would filter other inscriptions (as they have same signing key), so we will not see adopted at all.
2. With same gas prices + signing key + payload + same parent_id even tx_id can be the same in the mempool, meaning same tx, deduped in mempool. With different gas prices, different behavior. Different signing keys unify the API in this case better at the moment (until tke is implemented).

I think this is confusing for the user regarding the API, but also I am not against it fully, so wanted feedback from both of you guys.
To me the decentralized seq -> different signing keys (must be supported),
while distributed seq -> the same signing keys -> horizontally scaled "same" sequencer.

I just noticed round-robin implementation so pls let's align here: https://discord.com/channels/973324189794697286/1502951288932012042/1502951292639641763.

**Review by hansieodendaal — APPROVED** [review]
LGTM

**Review by davidrusu — APPROVED** [review]
Thanks for going through so many versions on this, it's looking much better
